### PR TITLE
Resolve the product code build warnings

### PR DESCRIPTION
**Pull request by xuzhg**
### Description

* Try to resolve the parts of build warnings as below:

image

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Comment by xuzhg** [comment]
The CI is passed at: https://identitydivision.visualstudio.com/OData/_build/results?buildId=521712&view=results

File: src/Microsoft.OData.Client/DataServiceContext.cs [hunk 1]
```diff
@@ -932,25 +934,25 @@ public void CancelRequest(IAsyncResult asyncResult)
...
-                        if (null != provider)
+                        if (provider != null)
                         {
                             context = provider.Context;
                         }
                     }
                 }
 
-                if (this != context)
+                if (context != null)
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/DataServiceContext.cs:955]: @xuzhg `if (this != context)` changed to `if (context != null)`. Kindly verify this change...

Comment (xuzhg) [reply src/Microsoft.OData.Client/DataServiceContext.cs:955]: @gathogojr Thanks, it's a typo. I fixed that with a force push.

File: src/Microsoft.OData.Client/BaseAsyncResult.cs [hunk 2]
```diff
@@ -115,14 +115,14 @@ public object AsyncState
...
+        public WaitHandle AsyncWaitHandle
         {
             get
             {
-                if (null == this.asyncWait)
+                if (this.asyncWait == null)
                 {   // delay create the wait handle since the user may never use it
                     // like asyncWait which will be GC'd, the losers in creating the asyncWait will also be GC'd
-                    System.Threading.Interlocked.CompareExchange(ref this.asyncWait, new System.Threading.ManualResetEvent(this.IsCompleted), null);
+                    Interlocked.CompareExchange(ref this.asyncWait, new System.Threading.ManualResetEvent(this.IsCompleted), null);
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/BaseAsyncResult.cs:125]: @xuzhg (Trivial) Drop `System.Threading` in `System.Threading.ManualResetEvent`

File: src/Microsoft.OData.Client/SaveResult.cs [hunk 3]
```diff
@@ -476,12 +476,12 @@ private ODataRequestMessageWrapper CreateNextRequest()
...
 
                     if (((EntityStates.Unchanged == descriptor.State) || (EntityStates.Modified == descriptor.State)) &&
-                        (null != (requestMessage = this.CheckAndProcessMediaEntryPut(entityDescriptor))))
+                        ((requestMessage = this.CheckAndProcessMediaEntryPut(entityDescriptor)) != null))
                     {
                         this.streamRequestKind = StreamRequestKind.PutMediaResource;
                         descriptorForSendingRequest2 = entityDescriptor.DefaultStreamDescriptor;  // We want to give SendingRequest2 the StreamDescriptor
                     }
-                    else if ((EntityStates.Added == descriptor.State) && (null != (requestMessage = this.CheckAndProcessMediaEntryPost(entityDescriptor))))
+                    else if ((EntityStates.Added == descriptor.State) && ((requestMessage = this.CheckAndProcessMediaEntryPost(entityDescriptor)) != null))
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/SaveResult.cs:484]: @xuzhg I think this `(EntityStates.Added == descriptor.State)` should change to `(descriptor.State == EntityStates.Added)`

File: src/Microsoft.OData.Client/SaveResult.cs [hunk 4]
```diff
@@ -476,12 +476,12 @@ private ODataRequestMessageWrapper CreateNextRequest()
                     EntityDescriptor entityDescriptor = (EntityDescriptor)descriptor;
 
                     if (((EntityStates.Unchanged == descriptor.State) || (EntityStates.Modified == descriptor.State)) &&
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/SaveResult.cs:478]: @xuzhg I think this
```
(EntityStates.Unchanged == descriptor.State) || (EntityStates.Modified == descriptor.State)
```
should change to
```
(descriptor.State == EntityStates.Unchanged) || (descriptor.State == EntityStates.Modified)
```

File: src/Microsoft.OData.Client/EntityTracker.cs [hunk 5]
```diff
@@ -246,7 +246,7 @@ internal override void AttachLink(object source, string sourceProperty, object t
                     case MergeOption.PreserveChanges:
                         if ((EntityStates.Added == existing.State) ||
                             (EntityStates.Unchanged == existing.State) ||
-                            (EntityStates.Modified == existing.State && null != existing.Target))
+                            (EntityStates.Modified == existing.State && existing.Target != null))
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/EntityTracker.cs:249]: @xuzhg (Trivial) Switch order of enum operands of == operator

File: src/Microsoft.OData.Client/SaveResult.cs (hunk 3 above)
Comment (xuzhg) [reply src/Microsoft.OData.Client/SaveResult.cs:484]: Changed.

File: src/Microsoft.OData.Client/SaveResult.cs (hunk 4 above)
Comment (xuzhg) [reply src/Microsoft.OData.Client/SaveResult.cs:478]: Changed. Thanks

File: src/Microsoft.OData.Client/EntityTracker.cs (hunk 5 above)
Comment (xuzhg) [reply src/Microsoft.OData.Client/EntityTracker.cs:249]: changed. Thanks.

**Pull request merged**